### PR TITLE
Various Doc Enhancements

### LINKED ISSUE #109: [Question/Feedback]: Resource naming to follow CAF abbreviation best practices
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

Initially, great job, I feel AVM wil really take off quickly. I only just learnt about it via the recently ALZ community call and already poking around anything with AVM in the git repos now. So, thank you.

Regarding naming, within our company we were trying to adhere to the CAF best practices [Abbreviation examples for Azure resources](https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations), and we were doing that via [terraform-azurerm-naming](https://github.com/Azure/terraform-azurerm-naming). I'm hoping that something like this could be the standard across the board in some way. I say this as I was looking at the [AVM Terraform storage module](https://github.com/Azure/terraform-azurerm-avm-res-storage-storageaccount/tree/main) and noticed that was using:

`name = "storage${random_pet.this.id}"`

Vs the CAF abbreviation. Thought this is possibly something that could be baked into AVM to make sure resources are consistent.

Again, awesome job on this, many thanks!

### LINKED ISSUE #115: [Question/Feedback]: Add note around deployment of dependant resources for SNFR2
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr2---category-testing---e2e-testing

Make note that examples should also deploy dependant resources

### LINKED ISSUE #130: [Feedback]: Explicit guidance on incorporating existing modules into AVM & on leveraging existing modules for inspiration to build net new AVM modules
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

Could we please get explicit guidance on incorporating existing modules into AVM incase there exist any instead of proposing and creating a new module

Also in case we propose and create a new module then could we get guidance on leveraging existing modules for inspiration to build net new AVM modules

### PR DESCRIPTION
**Pull request by jtracey93**
# Overview/Summary

Various Doc Enhancements

## This PR fixes/adds/changes/removes

1. Fixes #130 
2. Fixes #109 
3. Fixes #115
4. Adds clarity around required resources and their deployment for testing to complete
5. Fixes Missing Help & Support link
6. Improves navigation between FAQ & Contribution Q&A
7. Improve pattern module naming clarity for Terraform
8. Add tfenv to TF tools
9. Add notes about manual review before `v0.1.0` releases until AVM Compliance Tests are available
10. Various spelling and grammar 

### Breaking Changes

None

## As part of this Pull Request I have

- [x] Read the Contribution Guide and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Verified-Modules/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Verified-Modules/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Verified-Modules/)
- [x] Ensured PR tests are passing
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: docs/content/specs/shared/_index.md [hunk 1]
```diff
@@ -249,6 +273,53 @@ To see a directory and file structure for a module, see the language specific co
...
+  sku                 = "PerGB2018"
+  retention_in_days   = 30
+}
+```
+
+###### Bicep
+
+```bicep
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2021-12-01-preview' = {
+  name: 'rsg-test-001'
```
Comment (jtracey93) [inline docs/content/specs/shared/_index.md:312]:
```suggestion
  name: 'law-test-001'
```

**Review by matebarabas — APPROVED** [review]

File: docs/content/contributing/q-and-a.md [hunk 2]
```diff
@@ -106,14 +112,27 @@ At this moment, only Microsoft FTEs can be module owners.
...
 - [Terraform contribution guide][TerrafromContribution]
 
-Feel free to reach to [AVMCore team][AVMCoreTeam] in case that additional help is needed.
+Feel free to reach out to the [AVM Core team][AVMCoreTeam] in case that additional help is needed.
+
+---
+
+### What do I do about existing modules that are available doing a similar thing to my module that I am proposing to develop and release?
+
+We suggest that you first review the FAQ of [What is happening to existing initiatives like CARML and TFVM?][CARMLTFVMEvolution] to understand AVMs intent for existing modules from known Microsoft projects etc.
```
Comment (AlexanderSehr) [inline docs/content/contributing/q-and-a.md:128]:
```suggestion
We suggest that you first review the FAQ of [What is happening to existing initiatives like CARML and TFVM?][CARMLTFVMEvolution] to understand AVM's intent for existing modules from known Microsoft projects etc.
```

File: docs/content/contributing/q-and-a.md [hunk 3]
```diff
@@ -106,14 +112,27 @@ At this moment, only Microsoft FTEs can be module owners.
...
+
+As part of the [Module Proposal process][ProcessOverview], the AVM core team will work with you to triage your proposal and also to ensure there are no existing modules, from known Microsoft projects, to migrate to AVM.
```
Comment (AlexanderSehr) [inline docs/content/contributing/q-and-a.md:130]:
```suggestion
As part of the [Module Proposal process][ProcessOverview], the AVM core team will work with you to triage your proposal. We also want to make sure that no similar existing modules from known Microsoft projects are already on their way to be migrated to AVM.
```

File: docs/content/contributing/q-and-a.md [hunk 4]
```diff
@@ -106,14 +112,27 @@ At this moment, only Microsoft FTEs can be module owners.
...
+
+- If there aren't any, then you can proceed with developing your module from scratch once given approval to proceed by the AVM core team.
+- However, if there are existing modules from Microsoft projects we would invite you to help us complete the migration to AVM of this module; this may also entail working with the existing module owner/team.
+
+For existing modules that may not be directly owned and developed by Microsoft or their employees you should first review the license applied to the GitHub repository hosting the module and understand its terms and conditions. More information on GitHub repositories and licenses can be found here in [Licensing a repository][GitHubLicensing]. Most modules will use a license that will allow you to take inspiration and copy all or parts from the module source code however, to confirm, you should always check the license and any conditions you may have to meet by doing this.
```
Comment (AlexanderSehr) [inline docs/content/contributing/q-and-a.md:135]:
```suggestion
For existing modules that may not be directly owned and developed by Microsoft or their employees you should first review the license applied to the GitHub repository hosting the module and understand its terms and conditions. More information on GitHub repositories and licenses can be found here in [Licensing a repository][GitHubLicensing]. Most modules will use a license that will allow you to take inspiration and copy all or parts from the module source code. However, to confirm, you should always check the license and any conditions you may have to meet by doing this.
```

File: docs/content/contributing/q-and-a.md [hunk 5]
```diff
@@ -123,7 +142,7 @@ To get list of labels that **MUST** be created on gitHub repositories where modu
 
 You **SHOULD NOT** use any additional labels.
 
-There is also [PowerShell script][MandatoryLabels] The [AVM core team][AVMCoreTeam] created that can help to apply those labels to module GiHub repository.
+There is also a [PowerShell script][MandatoryLabels] that the [AVM core team][AVMCoreTeam] created that can help to apply those labels to module GiHub repository.
```
Comment (AlexanderSehr) [inline docs/content/contributing/q-and-a.md:145]:
```suggestion
There is also a [PowerShell script][MandatoryLabels] that the [AVM core team][AVMCoreTeam] created that can help to apply those labels the GitHub module repository.
```

File: docs/content/contributing/terraform/_index.md [hunk 6]
```diff
@@ -61,7 +61,8 @@ The following tooling/extensions are recommended to assist you developing for th
 
 - [CodeTour extension for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=vsls-contrib.codetour)
 - [Go (for writing tests)](https://go.dev/doc/install)
-- [Go extenstion for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=golang.go)
+- [Go extension for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=golang.go)
```
Comment (AlexanderSehr) [inline docs/content/contributing/terraform/_index.md:64]:
```suggestion
- [Extension for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=golang.go)
```

Comment (AlexanderSehr) [reply docs/content/contributing/terraform/_index.md:64]: Should we propose extension?

Comment (AlexanderSehr) [reply docs/content/contributing/terraform/_index.md:64]: Likewise, for the bracketpair instructions below, the tip will only work if the matching extension is installed

File: docs/content/specs/shared/_index.md [hunk 7]
```diff
@@ -249,6 +273,53 @@ To see a directory and file structure for a module, see the language specific co
...
+It is likely that to complete E2E tests, a number of resources will be required as dependencies to enable the tests to pass successfully. Some examples:
+
+- When testing the Diagnostic Settings interface for a Resource Module, you will need an existing Log Analytics Workspace to be able to send the logs to as a destination.
+- When testing the Private Endpoints interface for a Resource Module, you will need an existing Virtual Network, Subnet and Private DNS Zone to be able to complete the Private Endpoint deployment and configuration.
+
+Module owners **MUST**:
+
+- Create the required resources that their module depends upon in the test file/directory
+  - They **MUST** use simple/native resource declarations/definitions in their respective IaC language
+  - They **MUST NOT** use other modules to deploy these required resources
```
Comment (AlexanderSehr) [inline docs/content/specs/shared/_index.md:287]: Not even published ones?
I'd recommend to reword this to "They must not use other local module references (...)" or something like that

File: docs/content/specs/shared/_index.md [hunk 8]
```diff
@@ -782,7 +868,7 @@ Example: `avm-res-compute-virtualmachine`
 
 #### ID: RMNFR2 - Category: Inputs - Parameter/Variable Naming
 
-A module **SHOULD** use the following standard inputs:
+A module **MUST** use the following standard inputs:
```
Comment (AlexanderSehr) [inline docs/content/specs/shared/_index.md:871]: Why would we enforce location? Some resources don't have that input, or require a hard-coded "global".

File: docs/content/specs/shared/_index.md (hunk 7 above)
Comment (jtracey93) [reply docs/content/specs/shared/_index.md:287]: Nope we as we discussed yesterday on the office hours. Keeping tests supper simple and native declared is best as it's way easier to troubleshoot

File: docs/content/specs/shared/_index.md [hunk 9]
```diff
@@ -222,6 +222,30 @@ This section includes **shared, non-functional requirements (SNFR)** for Bicep a
 
 <br>
 
+#### ID: SNFR25 - Category: Composition - Resource Naming
+
+Module owners **MUST** set the default resource name prefix for child, extension, and interface resources to the associated abbreviation for the specific resource as documented in the following CAF article [Abbreviation examples for Azure resources](https://learn.microsoft.com/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations), if specified and documented. This reduces the amount of input values a module consumer **MUST** provide by default when using the module.
```
Comment (AlexanderSehr) [inline docs/content/specs/shared/_index.md:227]: Hey @matt-FFFFFF, just found this little nugget which requires updates on the Bicep side. Also cross-checked (as I wanted to use your implementation for PE also for Bicep) and saw you're using `pe` while the linked article proposes `pep`. I usually also always used `pe`, but I wonder if that causes confusion if we point something else out here. Thoughts?

Comment (jtracey93) [reply docs/content/specs/shared/_index.md:227]: Yup we must use what's in the doc as an abbreviation. I looked through both modules we've been working on and this is the only change I could see. 

Aligning to the doc keeps out of the way of an awkward conversation as to why we want to be different. 

Let's just align 👍

I'll happily put PR's in to fix

Comment (AlexanderSehr) [reply docs/content/specs/shared/_index.md:227]: Actually it's even written out as `pep` just 2 lines below

File: docs/content/specs/shared/_index.md (hunk 8 above)
Comment (jtracey93) [reply docs/content/specs/shared/_index.md:871]: Clarified
